### PR TITLE
[chalf] enable skipped tests

### PR DESCRIPTION
**Pull request by kshitij12345**
Ref: https://github.com/pytorch/pytorch/pull/79217#pullrequestreview-1002849962

Had to add a few `expectedFailures` for 
* reference test of `abs`,
* `executor` tests of `sigmoid`, `eq`, `masked_fill` (# NotImplementedError: argument of type: <class 'complex'>)
* `reference` and `executor` tests for `mul`, `pow`, `sub`, `true_divide`.

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -373,7 +370,6 @@ def _to_tensormeta(x):
             return x
 
         # TODO: iterate over requires_grad true/false
-        inps = tuple(op.reference_inputs(device, dtype, requires_grad=False))
```
Comment (kshitij12345) [inline test/test_ops.py:376]: unused variable

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -19949,6 +19949,16 @@ def __init__(
     ElementwiseUnaryPythonRefInfo(
         "_refs.abs",
         torch_opinfo_name="abs",
+        skips=(
+            # Reference result was farther (0.0) from the precise computation
+            # than the torch result was (nan)!
```
Comment (ngimel) [inline torch/testing/_internal/common_methods_invocations.py:19954]: interesting, do you know what inputs result in 0 vs nan?

**Review by ngimel — APPROVED** [review]

**Comment by ngimel** [comment]
Github doesn't allow to expand changes in common_methods_invocation. Can you list in the PR description what ops have to be skipped, and what are the inputs for 0 - nan difference?

File: test/test_ops.py [hunk 3]
```diff
@@ -361,9 +361,6 @@ def test_numpy_ref(self, device, dtype, op):
     @onlyNativeDeviceTypes
     @ops(python_ref_db)
     def test_python_ref_meta(self, device, dtype, op):
-        if dtype is torch.chalf:
```
Comment (mruberry) [inline test/test_ops.py:364]: Awesome!

File: test/test_ops.py [hunk 4]
```diff
@@ -394,9 +390,6 @@ def _to_tensormeta(x):
                         prims.utils.compare_tensor_meta(a, b)
 
     def _ref_test_helper(self, ctx, device, dtype, op, skip_zero_numel=False):
-        if dtype is torch.chalf:
```
Comment (mruberry) [inline test/test_ops.py:397]: Cool

**Review by mruberry — APPROVED** [review]
Cool! Updating the PR summary per @ngimel's request is a good idea, though, even after this merges

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py:19954]: Haven't checked. Will try to check them tomorrow and update here.

**Comment by kshitij12345** [comment]
Updated the PR description.

**Pull request closed without merging**